.github/workflows/build.yml

```diff
@@ -233,11 +201,84 @@ jobs:
           dotnet tool install --global dotnet-dump
           dotnet-dump analyze ${{ steps.detect-dump.outputs.DUMP_FILE }} -c "clrstack" -c "pe -lines" -c "exit"
 
+  package:
+    runs-on: windows-latest
+    needs: [build]
+    strategy:
+      fail-fast: false # prevent one matrix pipeline from being cancelled if one fails, we want them all to run to completion.
+      matrix:
+        winui: [2, 3]
+
+    env:
+      VERSION_PROPERTY: ${{ github.ref == 'refs/heads/main' && format('build.{0}', github.run_number) || format('pull-{0}.{1}', github.event.number, github.run_number) }}
+
+    steps:
+      - name: Install .NET SDK v${{ env.DOTNET_VERSION }}
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+
+      - name: .NET Info (if diagnostics)
+        if: ${{ env.ENABLE_DIAGNOSTICS == 'true' }}
+        run: dotnet --info
+
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
```

Comment:
We have a bunch of these shared steps between the build and package phase, I think there's a way we can specify them in a separate YAML file and then include them? That way we can ensure they're consistent.

https://docs.github.com/en/actions/using-workflows/avoiding-duplication - think maybe a re-usable workflow could work, I'm not sure if I understand if a job can use a workflow and then add steps though... doesn't seem like it, boo. Setting up a composite action for this seems like a pain... Thoughts?

Reply:
I agree, it's a bit redundant. I did some light work with reducing the redundancy but nothing in-depth, I would save that for another PR.